Merge Request

omerpalaz opened the pull request.

omerpalaz committed:
First test commit.

omerpalaz committed:
README.md file update.

omerpalaz committed:
First test commit.

omerpalaz committed:
Message class is added.

omerpalaz committed:
"LsPrefix" message type is added.

omerpalaz committed:
"LsLink" message type class is added.

omerpalaz committed:
"LsNode" message type class is added.

omerpalaz committed:
"LsNode" message type class is added.

omerpalaz committed:
"BmpStat" and "BaseAttribute" message type class is added.

omerpalaz committed:
"Peer" message type class is added.

omerpalaz committed:
"Collector" and "Router" message type class is added.

omerpalaz committed:
- "Example_Kafka_Consumer" is added for test purposes.
- Message type classes are put in folder "Message_Types"
- "Message" class is debugged.

omerpalaz committed:
- Parser in "Base" class completed.

omerpalaz committed:
- 'toJson' and 'toJsonPretty' functions are implemented.

omerpalaz committed:
- 'FieldProcessors' class is added.
- 'Kafka_Consumer' class is added to test the api.

omerpalaz committed:
- Comments are added to the code.

omerpalaz committed:
- API is a package to use now.

omerpalaz committed:
- README.md file is completed.
- Documentation of "getMapRow()" is added to the class 'Base'

omerpalaz committed:
README.md is updated.

omerpalaz committed:
- Message type classes are debugged.

TimEvens commented on .idea/vcs.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can you remove the .idea directory.  Can you also add a .gitignore similar to the other projects to prevent .idea and other unwanted files in the repo?

omerpalaz replied:
- ".idea" directory is removed.
- ".gitignore" is added to the project repo.

TimEvens commented on Kafka_Consumer.py:
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
```
Can you move Kafka_Consumer.py to examples/?   Can you also rename this to be "log_consumer.py"

omerpalaz replied:
- "examples" directory is created.
- "Kafka_Consumer.py" is renamed to be "log_consumer.py" in examples folder.

TimEvens commented on Kafka_Consumer.py:
```diff
@@ -0,0 +1,101 @@
+        print router.toJsonPretty()
+
+    elif t == "openbmp.parsed.peer":
+        peer = Peer(m.getContent())
```
let's have all the extend classes require "version" like router does.  This way we start it off requiring that. Alternatively, the first arg can be the message class object instead, which contains the version and content.

omerpalaz replied:
Each extend class takes Message object as first argument of object constructor.

TimEvens commented on README.md:
````diff
@@ -1,2 +1,61 @@
+### Code
+
+```python
+from api import *
````
Should be "openbmp.api.parsed.message"

TimEvens commented on README.md:
```diff
@@ -1,2 +1,61 @@
+msg = Message(raw_kafka_message_data)
+
+#Parse the content data by its type
+peer = Peer(msg.getContent());
```
Update to include version or just the message class object instead of msg.getContent().

omerpalaz replied:
Updated to version which includes new version of message class.

TimEvens commented on api/Base.py:
```diff
@@ -0,0 +1,96 @@
+
+    def __init__(self):
+        """Initializes the class variables."""
+        self.spec_version = float(0.0) # Configured message bus specification version (max) supported
```
Default to should based on latest, so this should be 1.3.

omerpalaz replied:
spec_version is initialized to 1.3

TimEvens commented on api/Base.py:
```diff
@@ -0,0 +1,96 @@
+            fieldsMap = dict(zip(self.headerNames, fields))
+
+            # Process and validate each field with its corresponding processor.
+            for (f, p, h) in zip(fields, self.getProcessors(), self.headerNames):
```
nice

TimEvens commented on api/BaseAttribute.py:
```diff
@@ -0,0 +1,70 @@
+
+        super(BaseAttribute, self).__init__()
+
+        self.headerNames = ["action", "seq", "hash", "router_hash", "router_ip", "peer_hash", "peer_ip",
```
Let's change all of these to use an enum/class for fields.  Moody is working on this change for Java as well where he has a fields schema class that defines all fields by code accessor which can then render the field name and value.   For example:

MsgBusFields.ACTION.getName() to get the header name and MsgBusFields.ACTION.getDefaultValue() to get the default value if not present in the TSV. We need this in code to avoid errors with people typo'ing the dictionary keys.  We also need this so that default values can be populated when supporting older schemas where the fields are not present.

TimEvens commented on api/LsLink.py:
```diff
@@ -0,0 +1,217 @@
+        super(LsLink, self).__init__()
+        self.spec_version = version
+
+        if version >= float(1.3):
```
We can change this to support adding to the list/array instead of duplicating.   Moody's changes also take this into account..

TimEvens commented on api/LsLink.py:
```diff
@@ -0,0 +1,217 @@
+
+        if self.spec_version >= float(1.3):
+
+            processors = [
```
Same thing for the processors.   Let's change it so that we only add the extra processors instead of duplicating it.

TimEvens commented on api/__init__.py:
```diff
@@ -0,0 +1,12 @@
+from Base import *
```
Normally we don't have an init like this because it causes an wanted load.   Let's remove that and require them to import directly.  Just leave this empty.

TimEvens commented:
All in all, good... but see my comments.  Talk to Moody about the changes he is making to support field names using enum/class and default values.     We need to ensure we are backwards compatible and that we support default values correctly.    


For the structure, can you change

**api/**

***TO***

**src/openbmp/api/parsed/message**

I didn't see a **setup.py**.   You should create one and validate that it works for:

* pip install 
* python setup.py install
* python setup.py bdist_wheel
* setup.py bdist_egg

Can you update the README.md to include setup/install instructions based on the wheel and manual using ```python setup.py install```

omerpalaz committed:
"api/" changed to "src/openbmp/api/parsed/message"

omerpalaz committed:
"api/" changed to "src/openbmp/api/parsed/message"

omerpalaz committed:
"api/" changed to "src/openbmp/api/parsed/message"

omerpalaz committed:
setup.py is created.

omerpalaz committed:
"__init__.py" file is edited.

omerpalaz committed:
"__init__.py" file is edited.

omerpalaz committed:
"__init__.py" file is edited.

omerpalaz committed:
.gitignore is added.

omerpalaz committed:
.gitignore is added.

omerpalaz committed:
.gitignore is added.

omerpalaz committed:
.gitignore is added.

omerpalaz committed:
.gitignore is added.

omerpalaz committed:
.gitignore fixed

omerpalaz committed:
.gitignore is added.

omerpalaz committed:
.gitignore is added.

omerpalaz committed:
- "log_consumer.py" is added.
- ".gitignore" is added.

omerpalaz committed:
- "log_consumer.py" is added.
- ".gitignore" is added.

omerpalaz committed:
- log_consumer is updated to support new version of message types
- Default schema version is 1.3

TimEvens commented:
A few changes:

* **setup.py** is empty.  It should be similar to what you see in the other python modules.  It needs to include the install for the openbmp items only. 
* README should be updated to include install steps, including wheel. 
* You are missing __init__.py under openbmp/, openbmp/api/, and openbmp/api/parsed. You need ```__init__.py``` under each module directory, otherwise import fails with no module found. 
* **examples/log_consumer.py** is not importing the library correctly.  Should be openbmp.api.parsed.message.  The  ```__all__``` variable is not defined in ```__init__.py``` , so it's not importing the needed modules.  Define __all__ so that ```import *``` works correctly. 
* When using ```import *``` you need to use <module>.<classname>.  For example, **Message.Message()**, not Message().   Alternatively, you can import each module/class (eg. from openbmp.api.parsed.message.Message import Message).

omerpalaz committed:
- BaseAttribute is compatible with java codebase.
- BmpStat is compatible with java codebase.

omerpalaz committed:
- Collector class is compatible with java codebase.

omerpalaz committed:
- LsLink class is compatible with java codebase.

omerpalaz committed:
- LsNode class is compatible with java codebase.

omerpalaz committed:
- LsPrefix class is compatible with java codebase.

omerpalaz committed:
- Peer class is compatible with java codebase.

omerpalaz committed:
- UnicastPrefix class is compatible with java codebase.

omerpalaz committed:
- setup.py  is created.

omerpalaz committed:
- README.md is updated with import of classes.
- setup.py is updated.

omerpalaz committed:
- README.md is updated with import of classes.

omerpalaz committed:
- README.md is updated with installation instructions.
 - Project directory structure fixed.

omerpalaz committed:
- README.md is updated with message type names.

omerpalaz committed:
- README.md is updated.

omerpalaz committed:
- setup.py is updated with correct license info.
- LICENSE file is added to the project.

omerpalaz committed:
- MsgBusFields uses objects instead of dict directly in order to support getName() and getDefaultValue() functions.
- config.yaml is added to example directory.
- log_consumer.py is updated to show timestamp, message type and version.

omerpalaz committed:
- README.md file is updated for installation instructions.

omerpalaz committed:
- log_consumer is updated to show "Received Message" text in logs.

omerpalaz committed:
- log_consumer is updated to show "Received Message" text in logs.
